src/BatteryStateOfChargeService.h

```diff
@@ -44,6 +44,14 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
 private:
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
-
+    double oldcurrent;
+    double newcurrent;
+    double totalcurrent;
+    QTime newtime;
+    QTime oldtime;
+    QTime et;
+    int timeused;
+    double s1;
+    double s2;
```

Comment:
A lot of these private members are only being used in the `addData` function. We want to keep the scope of variables as small as we can, so variables like s1, s2, should just be local variables inside the `addData` function.